contracts/ETO/ETOCommitment.sol

```diff
@@ -916,58 +927,57 @@ contract ETOCommitment is
-        uint256 newTotalTokensInt = _totalTokensInt + equityTokenInt;
-        assert(newTotalTokensInt < 2**56);
-        // as fixedSlotEquityTokenInt256 always lte equityTokenInt no chances to overflow
-        uint56 newTotalFixedSlotsTokensInt = _totalFixedSlotsTokensInt + uint56(fixedSlotEquityTokenInt);
+        // total number of tokens must fit into uint128, uints are coerced to biggest size
+        // as ticket token storage is 2**96, we need 2**32 tickets of max size to overflow so disregard
+        uint128 newTotalTokenAmount = uint128(_totalTokenAmount + equityTokenAmount);
+        // as fixedSlotEquityTokenAmount always lte equityTokenAmount no chances to overflow
+        uint128 newTotalFixedSlotsTokenAmount = _totalFixedSlotsTokenAmount + uint128(fixedSlotEquityTokenAmount);
         // add new investor
-        uint32 newTotalInvestors = _totalInvestors + (firstTimeInvestment ? 1 : 0);
-        // new total eur equivalent invested
-        uint256 newTotalEquivEurUlps = _totalEquivEurUlps + equivEurUlps;
-        assert(newTotalEquivEurUlps < 2**112);
-        // resue eth/neur contribution slots for native currency bookkeeping
-        uint112 newTotalEth = _additionalContributionEth;
-        uint112 newTotalEurUlps = _additionalContributionEurUlps;
+        uint128 newTotalInvestors = _totalInvestors + (firstTimeInvestment ? 1 : 0);
+        // new total eur equivalent invested, disregard overflow for the same reason as with tokens
+        uint128 newTotalEquivEurUlps = uint128(_totalEquivEurUlps + equivEurUlps);
+        // resue eth/neur contribution slots for native currency bookkeeping, no overflow if above does not overflows
+        uint128 newTotalEth = _additionalContributionEth;
+        uint128 newTotalEurUlps = _additionalContributionEurUlps;
         if (msg.sender == address(EURO_TOKEN)) {
-            newTotalEurUlps += uint112(amount);
+            newTotalEurUlps += uint128(amount);
         } else {
-            newTotalEth += uint112(amount);
+            newTotalEth += uint128(amount);
         }
 
         // write to storage in one go so maybe optimizer will work
-        _totalTokensInt = uint56(newTotalTokensInt);
-        _totalFixedSlotsTokensInt = newTotalFixedSlotsTokensInt;
+        _totalTokenAmount = newTotalTokenAmount;
+        _totalFixedSlotsTokenAmount = newTotalFixedSlotsTokenAmount;
         _totalInvestors = newTotalInvestors;
-        _totalEquivEurUlps = uint112(newTotalEquivEurUlps);
+        _totalEquivEurUlps = newTotalEquivEurUlps;
         _additionalContributionEth = newTotalEth;
         _additionalContributionEurUlps = newTotalEurUlps;
     }
 
-    function isCapExceeded(bool applyDiscounts, uint256 equityTokenInt, uint256 fixedSlotsEquityTokenInt, uint256 equivEurUlps)
+    function isCapExceeded(bool applyDiscounts, uint256 equityTokenAmount, uint256 fixedSlotsEquityTokenAmount, uint256 equivEurUlps)
```

Comment:
this could with some documentation maybe:

> returns whether or not the cap would be exceeded if another investment defined by the given parameters would be made:
@param equityTokenAmount total amount of new equityTokens to be added
@param fixedSlotsEquityTokenAmount tokens of the above would be associated to reserved/fixed/pre-allocated slots
@param equivEurUlps the amount of EURO_TOKEN that is equiv to the newly-created tokens

Reply:
thx for the docstring